app/models/compute_resources/foreman/model/libvirt.rb

```diff
@@ -262,6 +262,10 @@ def disconnect
       self.class.terminate_connection(url)
     end
 
+    def image_exists?(uuid)
+      File.exist?(uuid) && File.readable?(uuid)
```

Comment:
These are direct file operations but what about remote compute resources or image stores where the Foreman user doesn't have access?

I think this needs to go though libvirt itself.